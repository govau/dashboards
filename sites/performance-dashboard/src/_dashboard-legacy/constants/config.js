export const SCREEN_SM = 768;
export const DEFAULT_HERO_HEIGHT = 240;
