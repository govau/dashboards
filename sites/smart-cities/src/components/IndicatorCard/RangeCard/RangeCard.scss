@import '~style/main';

.wrapper {
  @include shadow;

  display: flex;
  flex-direction: column;
  position: relative; // .indicatorTypeMark is absolute, relative to this
  background: $color-white;
  border: none;
  border-radius: 4px;
  overflow: hidden; // clip the footer with the border radius
}

.smallCard {
  height: 112px;
}

.largeCard {
  height: 168px;
}

.indicatorName {
  flex: 1;
  position: absolute;
  right: 54px;
  font-weight: 500;

  .smallCard & {
    left: 24px;
    top: 3px;
    font-size: 12px;
    line-height: 16px;
  }

  .largeCard & {
    left: 29px;
    top: 4px;
    font-size: 16px;
    line-height: 1.1;
  }
}

.indicatorTypeMark {
  position: absolute;

  .smallCard & {
    top: 2px;
    left: 4px;
  }

  .largeCard & {
    top: 4px;
    left: 9px;
  }
}

.chartWrapper {
  margin: auto 3px 0;

  .smallCard & {
    height: 60px;
  }

  .largeCard & {
    height: 110px;
  }
}

.svg {
  width: 100%;
  height: 100%;
}

.bar {
  stroke: white;
}

.footer {
  display: flex;
  justify-content: space-between;
  padding: 5px;

  .smallCard & {
    height: 36px;
  }

  .largeCard & {
    height: 40px;
  }
}

.maxValues {
  text-align: right;
}

.footerCity,
.footerValue {
  .smallCard & {
    font-size: 10px;
    line-height: 12px;
  }

  .largeCard & {
    font-size: 12px;
    line-height: 14px;
  }
}

.footerValue {
  font-weight: 700;
}
