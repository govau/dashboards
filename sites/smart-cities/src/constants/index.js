export * from './misc';
export { default as INDICATORS } from './indicators';
export * from './categories';
export { default as CITIES } from './cities';
