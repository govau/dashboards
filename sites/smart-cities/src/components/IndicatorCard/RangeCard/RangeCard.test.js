import React from 'react';
import { shallow } from 'enzyme';
import RangeCard from './RangeCard';
import { CARD_SIZES } from '../../../constants';

jest.mock('../../IndicatorIcon/IndicatorIcon', () => 'IndicatorIcon');

const defaultProps = {
  colorName: 'jobs',
  cities: [
    {
      id: 'perth',
      name: 'Perth',
      indicators: {
        growthRate: 0.77,
      },
    },
    {
      id: 'sydney',
      name: 'Sydney',
      indicators: {
        growthRate: 0.22,
      },
    },
    {
      id: 'melbourne',
      name: 'Melbourne',
      indicators: {
        growthRate: 0.33,
      },
    },
  ],
  indicator: {
    id: 'growthRate',
    name: 'The population growth',
    format: '0[.]0%',
    contextual: false,
  },
  size: CARD_SIZES.LARGE,
};

it('should match Snapshot', () => {
  const component = shallow(<RangeCard {...defaultProps} />);
  expect(component.debug()).toMatchSnapshot();
});

it('should apply the largeCard class', () => {
  const component = shallow(
    <RangeCard
      {...defaultProps}
      size={CARD_SIZES.LARGE}
    />
  );

  expect(component.hasClass('largeCard')).toBe(true);
});

it('should apply the smallCard class', () => {
  const component = shallow(
    <RangeCard
      {...defaultProps}
      size={CARD_SIZES.SMALL}
    />
  );

  expect(component.hasClass('smallCard')).toBe(true);
});

it('should render the correct min and max info for specific cities', () => {
  const component = shallow(
    <RangeCard
      {...defaultProps}
      cities={[
        {
          id: 'perth',
          name: 'Perth',
          indicators: {
            growthRate: 0.77,
          },
        },
        {
          id: 'sydney',
          name: 'Sydney',
          indicators: {
            growthRate: 0.22,
          },
        },
        {
          id: 'melbourne',
          name: 'Melbourne',
          indicators: {
            growthRate: 0.33,
          },
        },
      ]}
    />
  );

  // min
  expect(component.find('.footerCity').at(0).text()).toBe('Sydney');
  expect(component.find('.footerValue').at(0).text()).toBe('22%');

  // max
  expect(component.find('.footerCity').at(1).text()).toBe('Perth');
  expect(component.find('.footerValue').at(1).text()).toBe('77%');
});

it('should handle multiple cities with the sane min and max values', () => {
  const component = shallow(
    <RangeCard
      {...defaultProps}
      cities={[
        {
          id: 'perth',
          name: 'Perth',
          indicators: {
            growthRate: 0.77,
          },
        },
        {
          id: 'act',
          name: 'Canberra',
          indicators: {
            growthRate: 0.22,
          },
        },
        {
          id: 'geelong',
          name: 'Geelong',
          indicators: {
            growthRate: 0.77,
          },
        },
        {
          id: 'sydney',
          name: 'Sydney',
          indicators: {
            growthRate: 0.22,
          },
        },
        {
          id: 'melbourne',
          name: 'Melbourne',
          indicators: {
            growthRate: 0.22,
          },
        },
      ]}
    />
  );

  // min
  expect(component.find('.footerCity').at(0).text()).toBe('3 cities');
  expect(component.find('.footerValue').at(0).text()).toBe('22%');

  // max
  expect(component.find('.footerCity').at(1).text()).toBe('2 cities');
  expect(component.find('.footerValue').at(1).text()).toBe('77%');
});

it('should render the correct number of bars', () => {
  const component = shallow(
    <RangeCard
      {...defaultProps}
      cities={[
        {
          id: 'perth',
          name: 'Perth',
          indicators: {
            growthRate: 0.77,
          },
        },
        {
          id: 'act',
          name: 'Canberra',
          indicators: {
            growthRate: 0.22,
          },
        },
        {
          id: 'geelong',
          name: 'Geelong',
          indicators: {
            growthRate: 0.77,
          },
        },
        {
          id: 'sydney',
          name: 'Sydney',
          indicators: {
            growthRate: 0.22,
          },
        },
        {
          id: 'melbourne',
          name: 'Melbourne',
          indicators: {
            growthRate: 0.22,
          },
        },
      ]}
    />
  );

  expect(component.find('.bar').length).toBe(5);
});

it('should exclude cities with no data', () => {
  const component = shallow(
    <RangeCard
      {...defaultProps}
      cities={[
        {
          id: 'perth',
          name: 'Perth',
          indicators: {}, // I have no data
        },
        {
          id: 'act',
          name: 'Canberra',
          indicators: {
            growthRate: 0.22,
          },
        },
        {
          id: 'geelong',
          name: 'Geelong',
          indicators: {
            growthRate: 0.77,
          },
        },
        {
          id: 'sydney',
          name: 'Sydney',
          indicators: {
            growthRate: 0.22,
          },
        },
        {
          id: 'melbourne',
          name: 'Melbourne',
          indicators: {
            growthRate: 0.22,
          },
        },
      ]}
    />
  );

  expect(component.find('.bar').length).toBe(4); // only 4, perth was hidden
});
