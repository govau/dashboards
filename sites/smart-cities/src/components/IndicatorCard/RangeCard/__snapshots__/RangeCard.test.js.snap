// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"wrapper largeCard\\">
  <p className=\\"indicatorName\\">
    The population growth
  </p>
  <IndicatorIcon className=\\"indicatorTypeMark\\" color=\\"JOBS_500\\" contextual={false} />
  <div className=\\"chartWrapper\\">
    <svg className=\\"svg\\">
      <rect className=\\"bar\\" x=\\"0%\\" y=\\"90.9090909090909%\\" width=\\"33.33333333333333%\\" height=\\"200%\\" rx=\\"2\\" ry=\\"2\\" fill=\\"JOBS_500\\" />
      <rect className=\\"bar\\" x=\\"33.33333333333333%\\" y=\\"72.72727272727272%\\" width=\\"33.33333333333333%\\" height=\\"200%\\" rx=\\"2\\" ry=\\"2\\" fill=\\"JOBS_060\\" />
      <rect className=\\"bar\\" x=\\"66.66666666666666%\\" y=\\"0%\\" width=\\"33.33333333333333%\\" height=\\"200%\\" rx=\\"2\\" ry=\\"2\\" fill=\\"JOBS_500\\" />
    </svg>
  </div>
  <div className=\\"footer\\" style={{...}}>
    <div>
      <div className=\\"footerCity\\">
        Sydney
      </div>
      <div className=\\"footerValue\\">
        22%
      </div>
    </div>
    <div className=\\"maxValues\\">
      <div className=\\"footerCity\\">
        Perth
      </div>
      <div className=\\"footerValue\\">
        77%
      </div>
    </div>
  </div>
</div>"
`;
